### PR TITLE
Skip PosTransactionsOrderRule for gateway nodes

### PR DESCRIPTION
**Pull request by noescape00**

### CODE REVIEW

File: src/Stratis.StratisD/Properties/launchSettings.json [hunk 1]
```diff
@@ -2,6 +2,7 @@
   "profiles": {
     "Stratis.StratisD": {
       "commandName": "Project",
+      "commandLineArgs": "-testnet -gateway"
```
Comment (zeptin) [inline src/Stratis.StratisD/Properties/launchSettings.json]: This should be reverted, please

Comment (noescape00) [reply src/Stratis.StratisD/Properties/launchSettings.json]: fixed

File: src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/PoSTransactionsOrderRule.cs [hunk 2]
```diff
@@ -8,12 +8,26 @@ namespace Stratis.Bitcoin.Features.Consensus.Rules.CommonRules
...
+            this.isGateway = this.Parent.ConnectionManagerSettings.IsGateway;
+        }
+
         /// <inheritdoc />
         /// <exception cref="ConsensusErrors.InvalidTxCount">Thrown in case block has less than 2 transaction.</exception>
         /// <exception cref="ConsensusErrors.PHCoinstakeMissmatch">Thrown in case coinstake transaction from the proven header missmatches 2nd block transaction.</exception>
         /// <exception cref="ConsensusErrors.ProofOfWorkTooHigh">Thrown in case block is a PoW block but created after the last pow height.</exception>
         public override void Run(RuleContext context)
         {
+            if (this.isGateway)
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/PoSTransactionsOrderRule.cs]: Why not just check `this.Parent.ConnectionManagerSettings.IsGateway` here?

**Comment by fassadlr** [comment]
@noescape00 so many tests failed! I'm re-running the build.

**Comment by fassadlr** [comment]
> @noescape00 so many tests failed! I'm re-running the build.

Wrong PR sorry.

**Comment by noescape00** [comment]
Closing this PR since it was decided that integrity rules work should be reverted and re-added in the next release

**Comment by noescape00** [comment]
Recreated in https://github.com/stratisproject/StratisBitcoinFullNode/pull/3726

**Pull request closed without merging**